File: c++代码/12.整数转罗马数字.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

class Solution
{
public:
    string intToRoman(int num)
    {
        map<int, string, greater<int> > roman_map;
        string ans_str = "";
        roman_map[1000] = "M";
        roman_map[900] = "CM";
        roman_map[500] = "D";
        roman_map[400] = "CD";
        roman_map[100] = "C";
        roman_map[90] = "XC";
        roman_map[50] = "L";
        roman_map[40] = "XL";
        roman_map[10] = "X";
        roman_map[9] = "IX";
        roman_map[5] = "V";
        roman_map[4] = "IV";
        roman_map[1] = "I";
        for (auto iter = roman_map.begin(); iter != roman_map.end(); ++iter)
        {
            while (num >= iter->first)
            {
                num -= iter->first;
                ans_str += iter->second;
            }
        }
        return ans_str;
    }
};

int main()
{
    Solution s;
    cout << s.intToRoman(1);
    return 0;
}

File: 力扣/字符串/12. 整数转罗马数字.py
class Solution:
    def intToRoman(self, num: int) -> str:
        #时间复杂度O(1),空间复杂度为O(1)
        d = {1000:'M',900:'CM', 500:'D', 400:'CD', 100:'C', 90:'XC', 50:'L', 40:'XL', 10:'X', 9:'IX', 5:'V', 4:'IV', 1:'I'}
        result = ''
        for i in d:
            while num >= i:
                num -= i
                result += d[i]
        return result
